packages/vuetify/src/components/VDataTable/VDataTable.ts

```diff
@@ -49,17 +49,13 @@ function searchTableItems (
   headersWithoutCustomFilters: DataTableHeader[],
   customFilter: DataTableFilterFunction
 ) {
-  let filtered = items
   search = typeof search === 'string' ? search.trim() : null
-  if (search && headersWithoutCustomFilters.length) {
-    filtered = items.filter(item => headersWithoutCustomFilters.some(filterFn(item, search, customFilter)))
-  }
-
-  if (headersWithCustomFilters.length) {
-    filtered = filtered.filter(item => headersWithCustomFilters.every(filterFn(item, search, defaultFilter)))
-  }
+  if (!(search && headersWithoutCustomFilters.length) && !headersWithCustomFilters.length) return items
 
-  return filtered
+  return items.filter(item =>
+    (headersWithCustomFilters.length && headersWithCustomFilters.every(filterFn(item, search, defaultFilter))) ||
```

Comment:
this should be broken up so it's easier to understand, possibly with some comments

Reply:
I broke up the callback a bit and added some comments so it's a bit easier to parse the flow of logic. Let me know if this is what you were going for or if you'd like something else 🙂